core/iwasm/common/wasm_shared_memory.c

```diff
@@ -87,6 +92,43 @@ search_module(WASMModuleCommon *module)
     return NULL;
 }
 
+static void
+create_list_of_waiter_addresses(void *key, void *value, void *user_data)
+{
+    /* add waiter address to atomic wait list */
+    AtomicWaitAddressNode *wait_address_node;
+
+    if (!(wait_address_node =
+              wasm_runtime_malloc(sizeof(AtomicWaitAddressNode)))) {
+        return;
+    }
+    wait_address_node->addr = key;
+    bh_assert(bh_list_insert(user_data, wait_address_node) == BH_LIST_SUCCESS);
+}
+
+void
+notify_stale_threads_on_exception(WASMModuleInstanceCommon *module_inst)
+{
+    bh_list atomic_wait_address_list_head;
+    bh_list *const atomic_wait_address_list = &atomic_wait_address_list_head;
+    static korp_mutex atomic_wait_address_list_lock;
```

Comment:
should the mutex be initialized?

Reply:
I actually don't think we need this mutex at all. What's the risk of calling this function from two different threads? Is it ok if we call `notify` twice on the same address in this specific instance?

Reply:
I was also skeptical about this, just thought would keep it for the draft and get a second opinion; will remove it.